eth/backend.go

```diff
@@ -321,32 +347,66 @@ func (gc *GoChain) SetEtherbase(etherbase common.Address) {
 	gc.miner.SetEtherbase(etherbase)
 }
 
-func (gc *GoChain) StartMining(local bool) error {
-	eb, err := gc.Etherbase()
-	if err != nil {
-		log.Error("Cannot start mining without etherbase", "err", err)
-		return fmt.Errorf("etherbase missing: %v", err)
-	}
-	if clique, ok := gc.engine.(*clique.Clique); ok {
-		wallet, err := gc.accountManager.Find(accounts.Account{Address: eb})
-		if wallet == nil || err != nil {
-			log.Error("Etherbase account unavailable locally", "err", err)
-			return fmt.Errorf("signer missing: %v", err)
+// StartMining starts the miner with the given number of CPU threads. If mining
+// is already running, this method adjust the number of threads allowed to use
+// and updates the minimum price required by the transaction pool.
+func (gc *GoChain) StartMining(threads int) error {
+	// Update the thread count within the consensus engine
+	type threaded interface {
+		SetThreads(threads int)
+	}
```

Comment:
Minor nit but `threaded` sounds like a boolean. I would suggest `threadSetter` as the interface name.

Reply:
I'm :+1: but I'm also hesitant to drift too far from upstream for minor changes since it creates so much noise when pulling new things in.